File: day1/part1.kt
import java.io.File

fun main() {
    var count = 0
    var curr = 50
    File("day1/input.txt").forEachLine { line ->
        val trimmed = line.trim()
        val dir = trimmed.take(1)
        val value = trimmed.filter{ it.isDigit() }.toInt()
        if (dir == "L") {
            curr -= value
            while (curr < 0) {
                curr += 100
            }
        } else if (dir == "R") {
            curr += value
            while (curr > 99) {
                curr -= 100
            }
        }
        if (curr == 0) {
            count++
        }
    }
    println(count)
}

File: day1/part1.py
curr = 50
count = 0
with open('input.txt', 'r') as file:
# with open('test.txt', 'r') as file:
    for line in file:
        line = line.strip()
        direction = line[0]
        value = int(line[1:])
        if direction == 'L':
            curr -= value
            while curr < 0:
                curr += 100
                # print(f"{value}: {curr}")
        elif direction == 'R':
            curr += value
            while curr > 99:
                curr -= 100
                # print(f"{value}: {curr}")
        if curr == 0:
            count += 1
print(count)